### ISSUE TITLE
DeleteUnregistered doesn't catch unregistered_class exceptions... (Trac #137)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#137]
it catches std::exception.  This should be more fine-grained.

Migrated from https://code.icecube.wisc.edu/ticket/137
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T21:08:15", 
    "description": "it catches std::exception.  This should be more fine-grained.", 
    "reporter": "troy", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547240895681654", 
    "component": "IceTray", 
    "summary": "DeleteUnregistered doesn't catch unregistered_class exceptions...", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2008-10-01T01:08:30", 
    "milestone": "", 
    "owner": "troy", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20090116T15:42:02: 
* troy changed status from "new" to "closed"
* troy changed resolution from "" to "fixed"
* troy commented:

> fixed in r24
> 
> now looks for unregistered_class and unsupported_version (unsupported_version only used by I3TrayInfo currently
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20141123T03:37:56: 
* olivas commented:

> Milestone offline-v3 deleted


* olivas changed milestone from "offline-v3" to ""

**Comment by icecube-trac** [comment]
Trac update at 20190111T18:54:57: nwhitehorn commented:

> In changeset 167886:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="167886"
> Fixes #170 and #137, with new params for Get and some docs
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190111T19:21:17: nwhitehorn commented:

> In changeset 168024:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168024"
> Fixes #170 and #137, with new params for Get and some docs
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190111T21:08:15: nwhitehorn commented:

> In changeset 168871:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168871"
> Fixes #170 and #137, with new params for Get and some docs
> ```
